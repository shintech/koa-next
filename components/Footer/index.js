import React from 'react'
import PropTypes from 'prop-types'
import Wrapper from './wrapper'

const Footer = ({ message = 'localhost' }) =>
  <Wrapper>
    <a href='#'>&#169;&nbsp;<span>{ message }</span></a>
  </Wrapper>

Footer.propTypes = {
  message: PropTypes.string.isRequired
}

export default Footer
