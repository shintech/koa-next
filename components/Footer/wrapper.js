import styled from 'styled-components'

const Wrapper = styled.footer`
  display: flex;
  min-height: 3vh;
  grid-row: 3;
  grid-column: 2;
  border-radius: 1ch;
  background-color: transparent;
  align-items: center;
  justify-content: center;

  a {
    padding: 1.25ch;
    margin-left: auto;
    margin-top: 0.5ch;
    color: orange;
    text-shadow: '0.08ch 0.08ch red';
    text-decoration: none;

    span {
      font-family: 'Rouge Script', cursive;
      font-size: 3ch;
      margin: 0;
    }
    
    &:hover {
      color: orangered;
      cursor: default;
    }    
  }

  svg {
    margin: 1vh 0 0 2vh;
    height: 5ch;
    width: 5ch;
  }
  
  @media (max-width: 800px) {
    grid-column: span 3;
  }  
`

export default Wrapper
