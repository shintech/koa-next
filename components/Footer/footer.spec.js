/* eslint-env jest */

import { shallow } from 'enzyme'
import { compose } from 'redux'
import toJSON from 'enzyme-to-json'
import Footer from 'components/Footer'

const shallowExpect = compose(expect, toJSON, shallow)

describe('COMPONENT -> Footer...', () => {
  let message = 'message'
  let footer = shallow(<Footer message={message} />)

  it(`expect Footer messsage text to equal "message"...`, () => {
    expect(footer.find('a span').text()).toBe(message)
  })
})

describe('COMPONENT -> Footer - snapshot...', () => {
  it('expect to render correct properties', () => {
    let message = 'message'

    shallowExpect(<Footer message={message} />).toMatchSnapshot()
  })
})
