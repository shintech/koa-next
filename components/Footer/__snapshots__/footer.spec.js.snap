// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT -> Footer - snapshot... expect to render correct properties 1`] = `
<wrapper__Wrapper>
  <a
    href="#"
  >
    © 
    <span>
      message
    </span>
  </a>
</wrapper__Wrapper>
`;
